### PR TITLE
feat: support UnixFSv1.5 metadata

### PR DESCRIPTION
**Pull request by achingbrain**
Adds [UnixFSv1.5 metadata](https://github.com/ipfs/specs/blob/master/UNIXFS.md#metadata) support.

Specifying metadata:

```console
$ jsipfs add --mode=0777 --mtime=100000 ./foo.txt
```

Respecting exsting metadata:

```console
$ jsipfs add --preserve-mode --preserve-mtime ./foo.txt
```

Displaying metadata:

```console
$ jsipfs ls Qmfoo
-rw-r--r-- Nov 21, 2019, 09:46:23 AM CST bar.txt Qbar 0
```

MFS gets `chmod` and `touch` commands:

```console
$ jsipfs files mkdir /foo
$ jsipfs files ls -l
drwxr-xr-x	-	foo	QmUNLLsPACCz1vLxQVkXqqLX5R1X345qqfHbsf67hvA3Nn	0
$ jsipfs files touch /foo
$ jsipfs files ls -l
drwxr-xr-x	Nov 27, 2019, 03:00:01 PM GMT	foo	QmVUPK8PbWDKw9v2PUMmJcCc5NhWXVwBawpi8E4Q4MHiCp	0
$ jsipfs files chmod 0777 /foo
$ jsipfs files ls -l
drwxrwxrwx	Nov 27, 2019, 03:00:01 PM GMT	foo	QmTa5rbJMxooh4euvwFt6UQAXQYi2AUg3sCmmd9AEpmipw	0
```

RPC Changes:

1. `/api/v0/files/chmod` and `/api/v0/files/touch` operations added
1. `mtime` and `mode` arguments added to `/api/v0/files/write` and `/api/v0/add`

Depends on:
 - [x] https://github.com/ipfs/js-ipfs-http-client/pull/1186
 - [x] https://github.com/ipfs/js-ipfs-mfs/pull/63
 - [x] https://github.com/ipfs/js-ipfs-multipart/pull/39
 - [x] https://github.com/ipfs/js-ipfs-unixfs/pull/40
 - [x] https://github.com/ipfs/js-ipfs-unixfs-importer/pull/42
 - [x] https://github.com/ipfs/js-ipfs-utils/pull/19
 - [x] https://github.com/ipfs/interface-js-ipfs-core/pull/572

### CODE REVIEW

**Comment by alanshaw** [comment]
sorry didn’t notice this was still in draft!

**Comment by achingbrain** [comment]
Yeah - just wanted to get this all somewhere that a) wasn't on my laptop and b) where we can talk about the API while the implementation settles (e.g. I write tests)

File: src/cli/parser.js [hunk 1]
```diff
@@ -36,6 +36,7 @@ const parser = yargs
   .commandDir('commands')
   .middleware(argv => Object.assign(argv, {
     getIpfs: utils.singleton(cb => utils.getIPFS(argv, cb)),
+    getStdin: () => process.stdin,
```
Comment (alanshaw) [inline src/cli/parser.js]: 👌

**Comment by achingbrain** [comment]
The pubsub tests will be the death of me.

**Pull request merged**